diff --git a/vllm/model_executor/layers/fused_moe/all2all_utils.py b/vllm/model_executor/layers/fused_moe/all2all_utils.py
--- a/vllm/model_executor/layers/fused_moe/all2all_utils.py
+++ b/vllm/model_executor/layers/fused_moe/all2all_utils.py
@@ -5,7 +5,6 @@ from typing import Any
 
 import torch
 
-from vllm.config import get_current_vllm_config
 from vllm.distributed import (
     get_ep_group,
 )
@@ -240,9 +239,7 @@ def maybe_make_prepare_finalize(
 
     elif moe.use_fi_nvl_one_sided_kernels:
         assert quant_config is not None
-        max_num_tokens = (
-            get_current_vllm_config().scheduler_config.max_num_batched_tokens
-        )
+        max_num_tokens = moe.max_num_tokens
         if quant_config.quant_dtype is None:
             dispatch_dtype_bytes_per_elem = 2
             dispatch_scale_bytes_per_token = 0
